code/__HELPERS/typeof.dm

```diff
@@ -118,4 +118,11 @@ var/global/list/existing_typesof_cache = list()
 	else
 		if(istype(A, path))
 			return TRUE
-	return FALSE
\ No newline at end of file
+	return FALSE
+//Finds types that are subtypes of a type, but only 1 level down.
+proc/direct_subtypesof(path)
+	var/list/out = subtypesof(path)
+	for(var/type in out)
+		for(var/stype in subtypesof(type))
```

Comment:
You can actually replace this second `for` loop by 
```dm
out -= subtypesof(type)
```

As BYOND overrides the `list1 - list2` operator between lists to mean "remove everything in list2 from list1".